File: templates/androidstarters-java/app/src/main/java/io/mvpstarter/sample/injection/component/AppComponent.java
package <%= appPackage %>.injection.component;

import android.app.Application;
import android.content.Context;

import javax.inject.Singleton;

import dagger.Component;
import <%= appPackage %>.data.DataManager;
import <%= appPackage %>.injection.ApplicationContext;
import <%= appPackage %>.injection.module.AppModule;

@Singleton
@Component(modules = AppModule.class)
public interface AppComponent {

    @ApplicationContext
    Context context();

    Application application();

    DataManager apiManager();
}

File: templates/androidstarters-kotlin/app/src/main/java/io/mvpstarter/sample/injection/component/AppComponent.kt
package <%= appPackage %>.injection.component

import android.app.Application
import android.content.Context
import dagger.Component
import <%= appPackage %>.data.DataManager
import <%= appPackage %>.data.remote.PokemonApi
import <%= appPackage %>.injection.ApplicationContext
import <%= appPackage %>.injection.module.AppModule
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AppModule::class))
interface AppComponent {

    @ApplicationContext
    fun context(): Context

    fun application(): Application

    fun dataManager(): DataManager

    fun pokemonApi(): PokemonApi
}
